js/modules/migrations/17/index.js

```diff
@@ -0,0 +1,55 @@
+const Message = require('../../types/message');
+
+
+exports.run = async (transaction) => {
+  const messagesStore = transaction.objectStore('messages');
+
+  console.log('Initialize messages schema version');
+  const numUpgradedMessages = await _initializeMessageSchemaVersion(messagesStore);
+  console.log('Complete messages schema version initialization', { numUpgradedMessages });
+
+  console.log('Create index from attachment schema version to attachment');
+  messagesStore.createIndex('schemaVersion', 'schemaVersion', { unique: false });
+};
+
+const _initializeMessageSchemaVersion = messagesStore =>
+  new Promise((resolve, reject) => {
+    const messagePutOperations = [];
+
+    const cursorRequest = messagesStore.openCursor();
+    cursorRequest.onsuccess = async (event) => {
+      const cursor = event.target.result;
+      const hasMoreData = Boolean(cursor);
+      if (!hasMoreData) {
+        await Promise.all(messagePutOperations);
+        return resolve(messagePutOperations.length);
+      }
+
+      const message = cursor.value;
+      const messageWithSchemaVersion = Message.initializeSchemaVersion(message);
+      messagePutOperations.push(putItem(
+        messagesStore,
+        messageWithSchemaVersion,
+        messageWithSchemaVersion.id
+      ));
+
+      return cursor.continue();
+    };
+
+    cursorRequest.onerror = event =>
+      reject(event.target.error);
+  });
+
+//    putItem :: IDBObjectStore -> Item -> Key -> Promise Item
+const putItem = (store, item, key) =>
+  new Promise((resolve, reject) => {
+    try {
+      const request = store.put(item, key);
+      request.onsuccess = event =>
+        resolve(event.target.result);
+      request.onerror = event =>
```

Comment:
You know what? I'm still confused as to the right way to capture errors from IndexedDB. Some docs tell me that it's right on the `transaction` or `request`, in the `error` key. And others use the method you've implemented here, trying to get it off of the `event`. Maybe you could try to test an error case, validate that this works the way you expect?

Reply:
My understanding that you can get it either from the `request` or `event` (inside a handler) as `event.target === request`. This is based on my usage of this as well as reviewing other libraries:
- `idb`: https://github.com/jakearchibald/idb/blob/306ec35f449b497dd0cc985f27ced70e0cda28a6/lib/idb.js#L10-L16
- `purescript-indexeddb`: https://github.com/truqu/purescript-indexedDB/blob/ef6b6498c356454f85b4c549ed4fd98b3d171944/src/Database/IndexedDB/IDBObjectStore.js#L1-L11